parser/ParseIndexProperties.hpp

```diff
@@ -0,0 +1,271 @@
+      return false;
+    }
+
+    for (const ParseKeyValue &key_value : *key_value_list) {
+      std::string key = ToLower(key_value.key()->value());
+      if (key.compare(kBloomFilterSizeInBytes) == 0) {
+        const ParseKeyIntegerValue *key_integer_value = static_cast<const ParseKeyIntegerValue*>(&key_value);
+        if (key_integer_value->value()->float_like()) {
+          setIndexDescriptionAsInvalid(&key_value, "size cannot be specified as a float");
+          return false;
+        }
+        const int64_t bloom_filter_size = key_integer_value->value()->long_value();
+        if (bloom_filter_size < 0) {
+          setIndexDescriptionAsInvalid(&key_value, "size cannot be negative");
+          return false;
+        }
+        // all good! allow the default bloom filter size to be modified
+        index_sub_block_description_->SetExtension(BloomFilterIndexSubBlockDescription::bloom_filter_size,
+                                                   bloom_filter_size);
+      } else if (key.compare(kBloomFilterNumHashes) == 0) {
+        const ParseKeyIntegerValue *key_integer_value = static_cast<const ParseKeyIntegerValue*>(&key_value);
+        if (key_integer_value->value()->float_like()) {
+          setIndexDescriptionAsInvalid(&key_value, "num_hashes cannot be specified as a float");
+          return false;
+        }
+        const int64_t number_of_hashes = key_integer_value->value()->long_value();
+        if (number_of_hashes < 0) {
+          setIndexDescriptionAsInvalid(&key_value, "num_hashes cannot be negative");
+          return false;
+        }
+        // all good! allow the default num_hashes to be modified
+        index_sub_block_description_->SetExtension(BloomFilterIndexSubBlockDescription::number_of_hashes,
+                                                   number_of_hashes);
+      } else if (key.compare(kBloomFilterProjectElementCount) == 0) {
+        const ParseKeyIntegerValue *key_integer_value = static_cast<const ParseKeyIntegerValue*>(&key_value);
+        if (key_integer_value->value()->float_like()) {
+          setIndexDescriptionAsInvalid(&key_value, "projected_element_count cannot be specified as a float");
+          return false;
+        }
+        const int64_t projected_element_count = key_integer_value->value()->long_value();
+        if (projected_element_count < 0) {
+          setIndexDescriptionAsInvalid(&key_value, "projected_element_count cannot be negative");
+          return false;
+        }
+        // all good! allow the default projected_element_count to be modified
+        index_sub_block_description_->SetExtension(BloomFilterIndexSubBlockDescription::projected_element_count,
+                                                   projected_element_count);
+      }
+    }
+    return true;
+  }
+};
+
+/**
+ * @brief Implementation of index properties for CSB Tree Index
+ */
+class CSBTreeIndexProperties : public IndexProperties {
+ public:
+  CSBTreeIndexProperties() {
```

Comment:
Most of these should be set in the initializer list similar to:
```c++
class Foo {
  Foo(int a) : a_(a) { }
 private:
  int a_;
};
```
This is because c++ will have already called default initializers by the time you enter the code block which is extra work and might cause unexpected behavior.